logstash-core/lib/logstash/environment.rb

```diff
@@ -29,6 +29,7 @@ module LogStash
             Setting::String.new("log.format", "plain", true, ["json", "plain"]),
             Setting::String.new("http.host", "127.0.0.1"),
               Setting::Port.new("http.port", 9600),
+            Setting::String.new("http.environment", "production"),
```

Comment:
@andrewvc Can we add a comment here why it was necessary and maybe pointing to this PR? I think in a few month we will wonders why we did such a thing.